drivers/gpio/gpio_mchp_xec_v2.c

```diff
@@ -0,0 +1,690 @@
+		    &gpio_xec_driver_api);
+
+static int gpio_xec_port040_076_init(const struct device *dev)
+{
+#if DT_IRQ_HAS_CELL(DT_NODELABEL(gpio_040_076), irq)
+	const struct gpio_xec_config *config = dev->config;
+
+	/* Turn on the block enable in the EC aggregator */
+	mchp_soc_ecia_girq_aggr_en(config->girq_id, 1);
+
+	IRQ_CONNECT(DT_IRQ(DT_NODELABEL(gpio_040_076), irq),
+		    DT_IRQ(DT_NODELABEL(gpio_040_076), priority),
+		    gpio_gpio_xec_port_isr,
+		    DEVICE_DT_GET(DT_NODELABEL(gpio_040_076)), 0U);
+
+	irq_enable(DT_IRQ(DT_NODELABEL(gpio_040_076), irq));
+#endif
+	return 0;
+}
+#endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(gpio_040_076), okay) */
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio_100_136), okay)
+static int gpio_xec_port100_136_init(const struct device *dev);
+
+static const struct gpio_xec_config gpio_xec_port100_136_config = {
+	.common = {
+		.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_DT_NODE(
+			DT_NODELABEL(gpio_100_136)),
+	},
+	.pcr1_base =
+		(uintptr_t) DT_REG_ADDR_BY_IDX(DT_NODELABEL(gpio_100_136), 0),
+	.parin_addr =
+		(uintptr_t) DT_REG_ADDR_BY_IDX(DT_NODELABEL(gpio_100_136), 1),
+	.parout_addr =
+		(uintptr_t) DT_REG_ADDR_BY_IDX(DT_NODELABEL(gpio_100_136), 2),
+	.port_num = MCHP_GPIO_100_136,
+#if DT_IRQ_HAS_CELL(DT_NODELABEL(gpio_100_136), irq)
+	.girq_id = 9,
+	.flags = GPIO_INT_ENABLE,
+#else
+	.flags = 0,
+#endif
+};
+
+static struct gpio_xec_data gpio_xec_port100_136_data;
+
+DEVICE_DT_DEFINE(DT_NODELABEL(gpio_100_136),
+		    gpio_xec_port100_136_init,
+		    NULL,
+		    &gpio_xec_port100_136_data, &gpio_xec_port100_136_config,
+		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,
+		    &gpio_xec_driver_api);
+
+static int gpio_xec_port100_136_init(const struct device *dev)
+{
+#if DT_IRQ_HAS_CELL(DT_NODELABEL(gpio_100_136), irq)
+	const struct gpio_xec_config *config = dev->config;
+
+	/* Turn on the block enable in the EC aggregator */
+	mchp_soc_ecia_girq_aggr_en(config->girq_id, 1);
+
+	IRQ_CONNECT(DT_IRQ(DT_NODELABEL(gpio_100_136), irq),
+		    DT_IRQ(DT_NODELABEL(gpio_100_136), priority),
+		    gpio_gpio_xec_port_isr,
+		    DEVICE_DT_GET(DT_NODELABEL(gpio_100_136)), 0U);
+
+	irq_enable(DT_IRQ(DT_NODELABEL(gpio_100_136), irq));
+#endif
+	return 0;
+}
+#endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(gpio_100_136), okay) */
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio_140_176), okay)
+static int gpio_xec_port140_176_init(const struct device *dev);
+
+static const struct gpio_xec_config gpio_xec_port140_176_config = {
+	.common = {
+		.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_DT_NODE(
+			DT_NODELABEL(gpio_140_176)),
+	},
+	.pcr1_base =
+		(uintptr_t) DT_REG_ADDR_BY_IDX(DT_NODELABEL(gpio_140_176), 0),
+	.parin_addr =
+		(uintptr_t) DT_REG_ADDR_BY_IDX(DT_NODELABEL(gpio_140_176), 1),
+	.parout_addr =
+		(uintptr_t) DT_REG_ADDR_BY_IDX(DT_NODELABEL(gpio_140_176), 2),
+	.port_num = MCHP_GPIO_140_176,
+#if DT_IRQ_HAS_CELL(DT_NODELABEL(gpio_140_176), irq)
+	.girq_id = 8,
+	.flags = GPIO_INT_ENABLE,
+#else
+	.flags = 0,
```

Comment:
it seems this code is heavily repeated ... would suggest making it more maintainable by reducing the repeated code ... you can do this using a macro if there really are no other options

Reply:
Done. Used DT FOREACH and a macro.